### PR TITLE
simplify setupContracts function

### PR DESCRIPTION
**Pull request by geoknee**
Rename it, make it synchronous, and add some TSDoc annotations!

This function is 
- poorly named (it's plural, when it shouldn't be)
- marked as async (returns a Promise) unnecessarily
- not self-explanatory (as to what it actually does)

This PR fixes all of these. It's a widely used test helper function, so the changeset is large. But it should be easy to understand the changeset by going commit-by-commit. 

---
### Code quality
- [x] I have written clear commit messages
- [x] I have performed a self-review of my own code
- [x] This change does not have an unduly wide scope
- [x] I have separated logic changes from refactor changes (formatting, renames, etc.)
- [x] I have commented my code wherever necessary (can be 0)
- [x] I have added tests that prove my fix is effective or that my feature works, if necessary
### Project management
- [x] I have applied the [appropriate labels](https://www.notion.so/Team-working-agreements-2a95c926bb5642e5a5c42e4b74a9dd24#b304e56734a74dfbb341b8b4b27b1c0c)
- [x] I have [linked to all relevant issues](https://help.zenhub.com/support/solutions/articles/43000010350-connecting-pull-requests-to-github-issues) (can be 0)
- [x] I have [linked to all dependent issues](https://help.zenhub.com/support/solutions/articles/43000010349-create-github-issue-dependencies) (can be 0)
- [x] I have assigned myself to this PR
- [x] I have chosen the appropriate [pipeline](https://www.notion.so/Team-working-agreements-2a95c926bb5642e5a5c42e4b74a9dd24#d534e68e7edc46fe8a4cda61b2258c4e) on zenhub for the linked issue

### CODE REVIEW

File: packages/nitro-protocol/test/contracts/ERC20AssetHolder/deposit.test.ts [hunk 1]
```diff
@@ -9,12 +9,12 @@ import {Channel, getChannelId} from '../../../src/contract/channel';
...
   getRandomNonce,
   getTestProvider,
-  setupContracts,
+  setupContract,
   writeGasConsumption,
 } from '../../test-helpers';
 
 const provider = getTestProvider();
-const signer0 = provider.getSigner(0); // Convention matches setupContracts function
+const signer0 = provider.getSigner(0); // Convention matchessetupContract function
```
Comment (NiloCK) [inline packages/nitro-protocol/test/contracts/ERC20AssetHolder/deposit.test.ts:17]:
```suggestion
const signer0 = provider.getSigner(0); // Convention matches setupContract function
```

File: packages/nitro-protocol/src/index.ts [hunk 2]
```diff
@@ -57,7 +57,7 @@ export {
   randomChannelId,
   randomExternalDestination,
   replaceAddressesAndBigNumberify,
-  setupContracts,
+  setupContract as setupContracts,
```
Comment (NiloCK) [inline packages/nitro-protocol/src/index.ts:60]: I see that this is part of the exported API of the `nitro-protocol` package.

Q: Who is consuming nitro-protocol's test helpers as a dependency?

**Review by NiloCK — APPROVED** [review]
Looks good. A definite improvement to the existing function.

File: packages/nitro-protocol/src/index.ts (hunk 2 above)
Comment (geoknee) [reply packages/nitro-protocol/src/index.ts:60]: Good question. I think probably no-one.

I don't think we ought to export it, really.

**Pull request merged**